core/time/src/clock.rs

```diff
@@ -185,14 +217,25 @@ impl FakeClock {
         if d <= Duration::ZERO {
             return;
         }
+        
         let receiver = {
             let mut inner = self.0.lock().unwrap();
+            let deadline = inner.now() + d;
+            
+            // Check if we should complete immediately
+            if inner.now() >= deadline {
+                return;
+            }
```

Comment:
This check seems unnecessary as in it will be quite rarely hit.  `deadline` is `now + d`.  And we know that `d` is not 0 so chances of this condition being true should be only when the process gets interrupted in between computing deadline and this line which should be quite rare.  So I would say we can remove it.